src/Mod/Sketcher/Gui/TaskSketcherTool.cpp

```diff
@@ -0,0 +1,967 @@
+        this, SLOT(emitSetparameterSix(double)));
+    this->installEventFilter(this);
+    ui->parameterOne->installEventFilter(this);
+    ui->parameterTwo->installEventFilter(this);
+    ui->parameterThree->installEventFilter(this);
+    ui->parameterFour->installEventFilter(this);
+    ui->parameterFive->installEventFilter(this);
+    ui->parameterSix->installEventFilter(this);
+    ui->notice->setWordWrap(true);
+    isWidgetActive = 0;
+}
+
+SketcherToolWidget::~SketcherToolWidget(){}
+
+//pre-select the number of the spinbox when it gets the focus.
+bool SketcherToolWidget::eventFilter(QObject* object, QEvent* event)
+{
+    if (object == ui->parameterOne && event->type() == QEvent::FocusIn) {
+        ui->parameterOne->selectNumber();
+    }
+    else if (object == ui->parameterTwo && event->type() == QEvent::FocusIn) {
+        ui->parameterTwo->selectNumber();
+    }
+    else if (object == ui->parameterThree && event->type() == QEvent::FocusIn) {
+        ui->parameterThree->selectNumber();
+    }
+    else if (object == ui->parameterFour && event->type() == QEvent::FocusIn) {
+        ui->parameterFour->selectNumber();
+    }
+    else if (object == ui->parameterFive && event->type() == QEvent::FocusIn) {
+        ui->parameterFive->selectNumber();
+    }
+    if (event->type() == QEvent::KeyPress) {
+        /*If a key shortcut is required to work on sketcher when a tool using Tool Setting widget
+        is being used, then you have to add this key to the below section such that the spinbox
+        doesn't keep the keypress event for itself. Note if you want the event to be handled by 
+        the spinbox too, you can return false.*/
+        QKeyEvent* keyEvent = static_cast<QKeyEvent*>(event);
+        if (keyEvent->key() == Qt::Key_Control)
+        {
+            sketchView->keyPressed(1, SoKeyboardEvent::LEFT_CONTROL);
+            return true;
+        }
+        else if (keyEvent->key() == Qt::Key_Shift)
+        {
+            sketchView->keyPressed(1, SoKeyboardEvent::RIGHT_SHIFT);
+            return true;
+        }
+        else if (keyEvent->key() == Qt::Key_Alt)
+        {
+            sketchView->keyPressed(1, SoKeyboardEvent::RIGHT_ALT);
+            return true;
+        }
+        else if (keyEvent->key() == Qt::Key_Escape)
+        {
+            sketchView->keyPressed(1, SoKeyboardEvent::ESCAPE);
+            return true;
+        }
+        else if (keyEvent->key() == Qt::Key_Space)
+        {
+            sketchView->keyPressed(1, SoKeyboardEvent::SPACE);
+            return true;
+        }
+    }
+    return false;
+}
+
+void SketcherToolWidget::emitTogglePreserveFilletChamferCorner(bool val) {
+    ui->checkBoxPreserveFilletChamferCorner->onSave();
+}
+void SketcherToolWidget::emitSetparameter(double val)
+{
+    isSettingSet[0] = 1;
+    toolParameters[0] = val;
+    if (unitChangeTriggeredValueChanged) {
+        unitChangeTriggeredValueChanged = 0;
+    }
+    else setParameterFocus(1);

```

Comment:
We do not want to have magic numbers in the code. It is difficult to read and error-prone. General solutions are constants or enums. 

I think I have left this in the code temporarily, but will need to be changed.